test/unit/spec/api/authSpec.js

```diff
@@ -48,182 +70,242 @@ define([
+         * 2. an error is returned from the request
+         * 3. the promise is resolved as an error
+         */
+        badTokenCases.forEach((badCase) => {
+            it(`${badCase.method} should get a 401 failure with a bad token`, () => {
+                mockAuthRequest(badCase.request, UNAUTH_ERROR, 401);
+                // the below is like calling:
+                // authClient.getTokenInfo(token)
+                // just slightly fancier
+                return authClient[badCase.method](...badCase.inputs)
+                    .then(() => {
+                        throw new Error('This should have failed!');
+                    })
+                    .catch((error) => {
+                        const request = jasmine.Ajax.requests.mostRecent();
+                        // expect that the token was applied here
+                        expect(request.requestHeaders.Authorization).toEqual(FAKE_TOKEN);
+                        expect(error).toBeDefined();
+                        expect(error.responseJSON).toBeDefined();
+                        expect(error.responseJSON).toEqual(UNAUTH_ERROR);
+                    });
+            });
         });
 
-        it('Should get auth token info', (done) => {
-            TestUtil.pendingIfNoToken();
-
-            authClient.getTokenInfo(token)
-                .then((response) => {
-                    expect(Object.keys(response)).toContain('expires');
-                    done();
-                })
-                .catch((error) => {
-                    expect(error).toBeNull();
-                    done();
-                });
+        /* test that running functions with missing tokens fail properly, such that
+         * 1. no token is attached to the Auth header
+         * 2. an error is returned from the request
+         * 3. the promise is resolved as an error
+         */
+        noTokenCases.forEach((badCase) => {
+            it(`${badCase.method} should get a 400 failure with no token`, () => {
+                clearToken();
+                mockAuthRequest(badCase.request, NO_TOKEN_ERROR, 400);
+                return authClient[badCase.method](...badCase.inputs)
+                    .then(() => {
+                        throw new Error('This should have failed!');
+                    })
+                    .catch((error) => {
+                        const request = jasmine.Ajax.requests.mostRecent();
+                        expect(request.requestHeaders.Authorization).toBeNull();
+                        expect(error).toBeDefined();
+                        expect(error.responseJSON).toBeDefined();
+                        expect(error.responseJSON).toEqual(NO_TOKEN_ERROR);
+                    });
+            });
         });
 
-        it('Should fail to get auth token info from a fake token', (done) => {
-            TestUtil.pendingIfNoToken();
+        it('Should make a new Auth client that has the expected functions', () => {
```

Comment:
Do you want to move this up to the top of the file, so you're first testing instantiation and that the object returned has the correct functions, and then testing the functionality?

Reply:
Sure, that makes sense for reading the tests. Done.